### DISCUSSION TITLE
Solweig - using meteorological dataset

### DISCUSSION THREAD

**Discussion by elidamd** [UMEP-dev/UMEP#427] [General]
Hi!

I want to share some strange behavior using SOLWEIG. (both GUI and processing tool).
My simulations using a file input for climate data returned Tmrt files with nodata for all the daytime hours.
I tested many changes that could be influencing the result, but the only valid output I could retrieve was manually setting the climate values for one single hour.

I could finally solve the problem using a climate dataset, once I checked the flag> "estimate diffuse and direct shortwave components from global radiation"

Given that my dataset contains the three inputs (global, diffuse, direct radiation), I wonder if this flag should not be mandatory for successful results. Is it an expected behavior from the tool?

Thanks!
UMEP version 3.19.3

**Comment by biglimp** [comment]
Please attach your meteorological file (or paste some line of your data here) for us to have a look at.

**Reply by elidamd** [reply]
The file I used is the extract (21st of December) from the typical meteorological year for Hamburg, retrieved from DWD.
https://www.dwd.de/DE/leistungen/testreferenzjahre/testreferenzjahre.html
[TRY2015_535299100078_Dec21_umep.txt](https://github.com/UMEP-dev/UMEP/files/8948311/TRY2015_535299100078_Dec21_umep.txt)

**Comment by biglimp** [comment]
Two things.

1. Your data cannot include hour=24. That does not exist. Hour goes fron 0 to 23
2. Your Kdir seams to be for a horisontal surface. It should be radiation perpendicular to the beam. See section 14.2.2.2 in the solweig manual.

Still you should not have nodata in your tmrt column. Maybe there is something strange with your geodata. Could you also attach your RunInfo-file so that we could chech thar the lat/lon is correct. What do you set as UTC?

**Reply by elidamd** [reply]
No problem! I am using climate data from DWD, the 24 hour instead of 0 is default in that file. 

I don't have the run-info file from the execution returning error anymore, but I used exactly the same geodata mentioned in the run-file I send attached. I traced the nodata failure, changing one parameter at a time until I could solve the problem. And the only setting that changed the result successfully was the flag I mentioned. So all information except for this will be the same in the file I send attached.
[RunInfoSOLWEIG_2015_355_0100.txt](https://github.com/UMEP-dev/UMEP/files/8979806/RunInfoSOLWEIG_2015_355_0100.txt)

**Reply by biglimp** [reply]
Hm, I see no issues in your RunInfo-file. Can you please try two other things. 1. Try using an anisotrophic sky
2. Update UMEP to 3.20
If that does not solve your issue, please upload all your input data for us to test if we can replicste you issue.
